db/disk/submission.go

```diff
@@ -323,3 +323,32 @@ func (this *backend) RemoveSubmission(assignment *model.Assignment, email string
 
     return true, nil;
 }
+
+func (this *backend) GetAttempts(assignment *model.Assignment, email string) ([]*model.GradingResult, error) {
+    submissions := make([]*model.GradingResult, 0);
+
+    submissionsDir := this.getUserSubmissionDir(assignment.GetCourse().GetID(), assignment.GetID(), email);
+    if !util.PathExists(submissionsDir) {
+        return submissions, nil;
+    }
+
+    dirents, err := os.ReadDir(submissionsDir);
+    if err != nil {
+        return nil, fmt.Errorf("Unable to read user submissions dir '%s': '%w'.", submissionsDir, err);
+    }
+
+    if len(dirents) == 0 {
```

Comment:
You can just remove this if block.
Since the loop will never be entered and the guard (range) is a trivial check, we can simplify a bit.